### ISSUE TITLE
Variadic functions don't seem to work

### ISSUE THREAD

**Issue by haukex** [Perl5-FFI/FFI-Platypus#210]
Hi,

Using libffi 3.2.1, Perl 5.28, FFI::Platypus 1.06, on Linux. Trying out the two examples for variadic functions in the documentation
```perl
use warnings;
use strict;
use FFI::Platypus;
my $ffi = FFI::Platypus->new( api => 1 );
$ffi->lib(undef);
$ffi->function( printf => ['string'] => ['int'] => 'int' )
    ->call("print integer %d\n", 42);
$ffi->function( printf => ['string'] => ['string'] => 'int' )
    ->call("print string %s\n", 'platypus');
```
gives random integers and random junk strings as output, different on each run, e.g.
```
print integer -254260256
print string �
```
Any help is appreciated!

**Comment by plicease** [comment]
This is definitely a bug.  I believe I have a fix, will need to do some testing first.

**Comment by plicease** [comment]
@haukex this should be fixed in 1.07.  Please let me know if you find otherwise.

**Comment by haukex** [comment]
Looks good, thanks very much for the quick response!
